pkg/reconcilermanager/controllers/rootsync_controller.go

```diff
@@ -416,6 +418,14 @@ func (r *RootSyncReconciler) deleteManagedObjects(ctx context.Context, reconcile
 
 // Register RootSync controller with reconciler-manager.
 func (r *RootSyncReconciler) Register(mgr controllerruntime.Manager, watchFleetMembership bool) error {
+	r.lock.Lock()
+	defer r.lock.Unlock()
```

Comment:
Do we need the lock? The `Register` function is only invoked once and won't be re-registered.

Reply:
The lock protects the controller from read/write race condition. But the way we're calling Register today it probably won't ever be called in parallel.

Reply:
We don't technically need it today, it's just good defensive practice to protect variables that could be used in parallel. 

Locking is cheap. Lock contention is what's expensive.